pkg/cache/cluster.go

```diff
@@ -1240,16 +1296,65 @@ func (c *clusterCache) processEvent(event watch.EventType, un *unstructured.Unst
 		return
 	}
 
+	c.eventMetaCh <- eventMeta{event, un}
+}
+
+func (c *clusterCache) processEvents() {
+	log := c.log.WithValues("fn", "processItems")
+	log.V(1).Info("Start processing events")
+
 	c.lock.Lock()
-	defer c.lock.Unlock()
-	existingNode, exists := c.resources[key]
-	if event == watch.Deleted {
-		if exists {
-			c.onNodeRemoved(key)
+	ch := c.eventMetaCh
+	c.lock.Unlock()
+
+	eventMetas := make([]eventMeta, 0)
+	ticker := time.NewTicker(c.eventProcessingInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case em, ok := <-ch:
+			if !ok {
+				log.V(1).Info("Event processing channel closed, finish processing")
```

Comment:
I think this can be debug level

Reply:
It completely makes sense. Fixed - https://github.com/argoproj/gitops-engine/commit/7a53ecab1d0eab6197dd11221f7c0b9ed3bed738